pkgs/by-name/ha/haredo/setup-hook.sh

```diff
@@ -0,0 +1,59 @@
+haredoBuildPhase() {
+    runHook preBuild
+
+    local buildTargets
+    read -ra buildTargets <<<"${haredoBuildTargets-}"
+    echoCmd "haredo build targets" "${buildTargets[@]}"
+    haredo ${enableParallelBuilding:+"-j${NIX_BUILD_CORES}"} "${buildTargets[@]}"
+
+    runHook postBuild
+}
+
+haredoCheckPhase() {
+    runHook preCheck
+
+    local checkTargets
+
+    if [[ -n "${haredoCheckTargets:-}" ]]; then
+        read -ra checkTargets <<<"${haredoCheckTargets}"
+    else
+        for dofile in "check.do" "test.do"; do
+            [[ -r "${dofile}" ]] && {
+                checkTargets=("${dofile%".do"}")
+                break
+            }
+        done
+    fi
+
+    if [[ -z "${checkTargets:-}" ]]; then
+        printf -- 'no check targets found for haredo\n'
+    else
+        echoCmd "haredo check targets" "${checkTargets[@]}"
+        haredo ${enableParallelChecking:+"-j${NIX_BUILD_CORES}"} "${checkTargets[@]}"
+    fi
+
+    runHook postCheck
+}
+
+haredoInstallPhase() {
+    runHook preInstall
+
+    local installTargets
+    read -ra installTargets <<<"${haredoInstallTargets:-"install"}"
+    echoCmd "haredo install targets" "${installTargets[@]}"
+    haredo ${enableParallelInstalling:+"-j${NIX_BUILD_CORES}"} "${installTargets[@]}"
```

Comment:
Can we enable parallel by default?

Reply:
Do you mean parallel by default only for the hook's `installPhase` or also for its `buildPhase` and `checkPhase`?

Reply:
I mean if it just works, just enable it for all like we do with cmake.

Reply:
I've taken a look at the [cmake hook](https://github.com/NixOS/nixpkgs/blob/ab3ea20adfe05a2328b4991612912ef7db67af83/pkgs/by-name/cm/cmake/setup-hook.sh#L128C16-L128C22) like you mention and changed `haredo`'s setup-hook behavior: now, to disable parallelism, the user must explicitly set the respective `enableParallel*` attribute to false. Any other case, that is, setting it to true or not setting it at all will enable parallelism.